packages/keystone/src/fields/types/float/index.ts

```diff
@@ -1,47 +1,162 @@
+// Float in GQL: A signed double-precision floating-point value.
+import { humanize } from '../../../lib/utils';
 import {
   BaseGeneratedListTypes,
   FieldTypeFunc,
   CommonFieldConfig,
-  fieldType,
   graphql,
+  fieldType,
   orderDirectionEnum,
-  FieldDefaultValue,
   filters,
 } from '../../../types';
+import { assertCreateIsNonNullAllowed, assertReadIsNonNullAllowed } from '../../non-null-graphql';
 import { resolveView } from '../../resolve-view';
 
 export type FloatFieldConfig<TGeneratedListTypes extends BaseGeneratedListTypes> =
   CommonFieldConfig<TGeneratedListTypes> & {
-    defaultValue?: FieldDefaultValue<number, TGeneratedListTypes>;
-    isRequired?: boolean;
+    defaultValue?: number;
     isIndexed?: boolean | 'unique';
+    isNullable?: boolean;
+    validation?: {
+      min?: number;
+      max?: number;
+      isRequired?: boolean;
+    };
+    graphql?: {
+      create?: {
+        isNonNull?: boolean;
+      };
+      read?: {
```

Comment:
This(specifically the read one) should only be allowed when `isNullable: false`, have a look at the integer field

Reply:
Can we talk on this? Using the patterns for this in integer led to failing tests, or failing typescript. I did this to make it a resolution